demos/blockfactory/factory_utils.js

```diff
@@ -926,16 +926,19 @@ FactoryUtils.sameBlockXml = function(blockXml1, blockXml2) {
   }
 
   // Compare the block elements directly. The XML tags may include other meta
-  // information we want to igrore.
+  // information we want to ignore.
   var blockElement1 = blockXml1.getElementsByTagName('block')[0];
   var blockElement2 = blockXml2.getElementsByTagName('block')[0];
 
   if (!(blockElement1 && blockElement2)) {
     throw new Error('Could not get find block element in XML.');
   }
 
-  var blockXmlText1 = Blockly.Xml.domToText(blockElement1);
-  var blockXmlText2 = Blockly.Xml.domToText(blockElement2);
+  var cleanBlockXml1 = FactoryUtils.cleanXml(blockElement1);
+  var cleanBlockXml2 = FactoryUtils.cleanXml(blockElement2);
+
+  var blockXmlText1 = Blockly.Xml.domToText(cleanBlockXml1);
+  var blockXmlText2 = Blockly.Xml.domToText(cleanBlockXml2);
 
   // Strip white space.
   blockXmlText1 = blockXmlText1.replace(/\s+/g, '');
```

Comment:
Hmm, this is a bug.  Not in your PR, but pre-existing.  It means that a text block (e.g. tooltip) with the value " " is the same as a text block with the value " &nbsp; &nbsp; &nbsp; ".

Is stripping whitespace even needed?

Reply:
Dug into it, it seems that the stripping of whitespace is covering a different bug.

The blockXml1 is the xml currently stored by the blockfactory. It is stored as pretty text to make it easy to export (I assume).

When you do:
* block/workspaceToDom
* domToPrettyText
* textToDom <- I think the DOM retains whitespace characters here
* domToText

You end up with pretty text at the end, instead of non-pretty text as you'd expect.

This means that the blockXml2, which goes:
* workspaceToDom
* domToText

is not equivalent.

I tried converting the blockXml2 into pretty text instead of normal text, but that doesn't work either (I assume something about the round tripping messes up the saved text's indentation).

There are a couple of options I've come up with:

1) Do a "direct" equality check by walking both dom trees and checking for attribute equality. This gets rid of the need to worry about whitespace (hopefully).
2) Store the block's xml as plain text/actual xml instead of pretty text. Convert it to pretty text only when getBlockLibrary is called.
3) Try to fix round tripping pretty text (probably better saved for another PR).

Or some combination of the three.